NewsApp/app/src/main/java/com/example/newsapp/features/auth/AuthViewModel.kt

```diff
@@ -1,8 +1,46 @@
 package com.example.newsapp.features.auth
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
+import com.example.newsapp.data.entities.User
 import com.example.newsapp.data.repos.UserRepository
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
 class AuthViewModel(
-    private val userRepository: UserRepository
-) : ViewModel()
\ No newline at end of file
+    private val userRepository: UserRepository,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) : ViewModel() {
+    private val registrationResultMutableLiveData = MutableLiveData<Boolean>()
+    val registrationResultLiveData: LiveData<Boolean>
+        get() = registrationResultMutableLiveData
+
+    private val loginResultMutableLiveData = MutableLiveData<Boolean>()
+    val loginResultLiveData: LiveData<Boolean>
+        get() = loginResultMutableLiveData
+
+    private val changePasswordMutableLiveData = MutableLiveData<Boolean>()
+    val changePasswordLiveData: LiveData<Boolean>
+        get() = changePasswordMutableLiveData
+
+    fun login(email: String, password: String) = viewModelScope.launch(ioDispatcher) {
+        loginResultMutableLiveData.postValue(userRepository.login(email, password))
+    }
+
+    fun register(user: User) = viewModelScope.launch(ioDispatcher) {
+        registrationResultMutableLiveData.postValue(userRepository.registerUser(user))
+    }
+
+    fun changePassword(email: String, password: String) = viewModelScope.launch(ioDispatcher) {
```

Comment:
I was thinking should we set a flag for the currently logged-in user to make our lives easier?
This way we'll know the email and the password of the user without having to request for them.